snuba/utils/streams/consumer.py

```diff
@@ -50,7 +50,11 @@ def poll(self, timeout: Optional[float] = None) -> Optional[Message[TPayload]]:
         raise NotImplementedError
 
     @abstractmethod
-    def commit(self) -> Mapping[Partition, int]:
+    def stage_offsets(self, offsets: Mapping[Partition, int]) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def commit_offsets(self) -> Mapping[Partition, int]:
         raise NotImplementedError
```

Comment:
This changes the behavior compared to the standard Kafka consumer. Unless it is already documented somewhere I am missing, I would advise to document the difference (no auto-commit and staging required before committing). 
I am assuming this is a behavior of the abstract consumer. Otherwise same comment applies to the KafkaConsumer.

Reply:
> This changes the behavior compared to the standard Kafka consumer.

This is not strictly true — it only changes the behavior of the _default configuration_ of the Confluent Kafka consumer. I agree it should be documented regardless, though. I'll do that.

Reply:
Added a lot of information about offset management here (part of this was pulled from old consumer code, part of it is new): a785b75205b6e399b09b17e0089d2ed01c1b708b

Let me know if there's anything that isn't addressed here that would be helpful to document.